File: 172.factorial-trailing-zeroes.0.go
/*
 * @lc app=leetcode id=172 lang=golang
 *
 * [172] Factorial Trailing Zeroes
 *
 * https://leetcode.com/problems/factorial-trailing-zeroes/description/
 *
 * algorithms
 * Easy (37.30%)
 * Total Accepted:    152.6K
 * Total Submissions: 409K
 * Testcase Example:  '3'
 *
 * Given an integer n, return the number of trailing zeroes in n!.
 *
 * Example 1:
 *
 *
 * Input: 3
 * Output: 0
 * Explanation: 3! = 6, no trailing zero.
 *
 * Example 2:
 *
 *
 * Input: 5
 * Output: 1
 * Explanation: 5! = 120, one trailing zero.
 *
 * Note: Your solution should be in logarithmic time complexity.
 *
 */
func trailingZeroes(n int) int {
	count := 0
	p := 5
	for n >= p {
		count += n / p
		p *= 5
	}
	return count
}

File: 172.factorial-trailing-zeroes.0.rs
/*
 * @lc app=leetcode id=172 lang=rust
 *
 * [172] Factorial Trailing Zeroes
 */
impl Solution {
    pub fn trailing_zeroes(n: i32) -> i32 {
        let num: i64 = n as i64;
        let mut c: i64 = 0;
        let mut p: i64 = 5;
        while num >= p {
            c += num / p;
            p *= 5;
        }
        c as i32
    }
}
